pkg/kubevirt/machine_server.go

```diff
@@ -100,6 +100,12 @@ func (p *MachinePlugin) DeleteMachine(ctx context.Context, req *driver.DeleteMac
 		return nil, util.PrepareErrorf(err, "Create machine %q failed on DecodeProviderSpecAndSecret", req.Machine.Name)
 	}
 
+	validationErrors := validation.ValidateKubevirtSecret(providerSpec, req.Secret)
+	if validationErrors != nil {
+		err = fmt.Errorf("error while validating ProviderSpec %v", validationErrors)
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
```

Comment:
The entire code snippet is repeated several times, always after `DecodeProviderSpecAndSecret`. Perhaps it could become part of the `DecodeProviderSpecAndSecret` function itself to eliminate the duplication?